py3status/modules/external_script.py

```diff
@@ -13,23 +13,52 @@
     cache_timeout: how often we refresh this module in seconds
         (default 15)
     format: see placeholders below (default '{output}')
+    format_notification: see placeholders below (default None)
     localize: should script output be localized (if available)
         (default True)
+    notification_state: specify notification state to use: current, click
+        (default 'current')
```

Comment:
@lasers I would suggest to make `click` the default here, right now notifications are appearing after each refresh (by default every 15 seconds) and I don't think this is a good default. 

Also, right now it's not possible to have notifications appearing _both_ after refresh and on click. I would suggest to replace `notification_state` with two new boolean options, `notify_after_refresh` (default `False`) and `notify_after_click` (default `True`)

Reply:
I agree `current` wasn't a good default. I made a new state `changed` that will print notifications only if it is changed. This would be a better default to use too. However, I added+consolidated more notification-related things so it is going to be an empty (nested) dictionary for default.

With this new nested dictionary, users can specify what notification should happen on `current`, `changed`, and `click` states along with (all) options: Levels, icons, message body or titles.

Although it may be over-engineered, I feel this approach is more clean than adding boolean + button configs that might not convey information well. Please check it out.

If you want to use all three states, you can... Just add them all in the dictionary, eg.
```python
    notifications = {
        'click': {'title': 'Click', 'msg': 'Clicking... {output_full}'},
        'current': {'title': 'Current', 'msg': 'Current... {output_full}'},
        'changed': {'title': 'Changed', 'msg': 'Changed... {output_full}'},
    }
```
If there are any other states that we can add, we should be able to add that too... rather than creating more new boolean + button configs.

Reply:
it's awesome now 👍